.github/workflows/weekly-tests.yaml

```diff
@@ -11,6 +11,87 @@ on:
+            - uses: actions/upload-artifact@v3
+              with:
+                  name: weekly-test-${{ github.run_number }}-attempt-${{ github.run_attempt }}-gem5-build-vega
+                  path: build/VEGA_X86/gem5.opt
+                  retention-days: 5
+            - run: echo "This job's status is ${{ job.status }}."
+
+    LULESH-tests:
+        runs-on: [self-hosted, linux, x64]
+        container: ghcr.io/gem5/gcn-gpu:latest
+        needs: build-gcn-gpu-gem5
+        timeout-minutes: 480 # 8 hours
+        steps:
+            - uses: actions/checkout@v3
+              with:
+                # Scheduled workflows run on the default branch by default. We
+                # therefore need to explicitly checkout the develop branch.
+                  ref: develop
+
+            - name: Download build/VEGA_X86/gem5.opt
+              uses: actions/download-artifact@v3
+              with:
+                  name: weekly-test-${{ github.run_number }}-attempt-${{ github.run_attempt }}-gem5-build-vega
+                  path: build/VEGA_X86
+                # `download-artifact` does not preserve permissions so we need to set
+                # them again.
+            - run: chmod u+x build/VEGA_X86/gem5.opt
+
+            - name: Obtain LULESH
+              # Obtains the latest LULESH compatible with this version of gem5 via
+              # gem5 Resources.
+              run: build/VEGA_X86/gem5.opt util/obtain-resource.py lulesh -p lulesh
+
+            - name: Run LULUESH tests
+              working-directory: ${{ github.workspace }}
+              run: |
+                  build/VEGA_X86/gem5.opt configs/example/apu_se.py -n3 --mem-size=8GB --reg-alloc-policy=dynamic --dgpu --gfx-version=gfx900 -c \
+                  lulesh --options="0.01 2"
+
+    HACC-tests:
+        runs-on: [self-hosted, linux, x64]
+        container: ghcr.io/gem5/gcn-gpu:latest
+        needs: build-gcn-gpu-gem5
+        timeout-minutes: 120 # 2 hours
+        steps:
+            - uses: actions/checkout@v3
+              with:
+          # Scheduled workflows run on the default branch by default. We
+          # therefore need to explicitly checkout the develop branch.
+                  ref: develop
+            - uses: actions/download-artifact@v3
+              with:
+                  name: weekly-test-${{ github.run_number }}-attempt-${{ github.run_attempt }}-gem5-build-vega
+                  path: build/VEGA_X86
+            - run: chmod u+x build/VEGA_X86/gem5.opt
+            - name: make hip directory
+              run: mkdir hip
+            - name: Compile m5ops and x86
+              working-directory: ${{ github.workspace }}/util/m5
+              run: |
+                  export TERM=xterm-256color
+                  scons build/x86/out/m5
+            - name: Download tests
+              working-directory: ${{ github.workspace }}/hip
+              run: wget http://dist.gem5.org/dist/v22-1/test-progs/halo-finder/ForceTreeTest
```

Comment:
Also wondering about this.  I doubt the binary itself has changed since 22.1, but I don't know if each version has duplicate copies on dist.gem5.org.  If not, changing the version would 404.

Reply:
So, the situation with the URLs is my old versioning system here (i.e., in this case `dist/<version>/test-progs` has basically broken down. I don't create a new sub-directory per-release. Each resources now has a and ID and a Version and we have a database which maps these (among other things, e.g., a link to documentation and source) to a URL.

So, as long as this version works for you it's fine. What we eventually need to do is add `ForceTreeTest` to our gem5 Resources database to make this mapping explicit. In this case we can use the "util/obtain-resource.py` tool to automatically get this resources (as we do with lulush).

@Harshil2107 : Is there any reason not to do this for ForceTreeTest? Can you do it quickly so we can just include it in this change?

Reply:
If this is where the binaries are, then I think this is fine and doesn't need to be immediately addressed.  But up to you @BobbyRBruce

Reply:
I can create a resource for the binary used in ForceTreeTest. Just to confirm the category would be binary?